app/waf/src/services/bigiq.service.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright 2019 F5 Networks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from '@loopback/service-proxy';
+import {inject, Provider} from '@loopback/core';
+import {BigIqDataSource} from '../datasources';
+import {factory} from '../log4ts';
+import {Logger} from 'typescript-logging';
+import {checkAndWait} from '../utils';
+
+const BIGIQ_HOST: string = process.env.BIGIQ_HOST || 'localhost';
+const BIGIQ_PORT: number = Number(process.env.BIGIQ_PORT) || 8888;
+const BIGIQ_USERNAME: string = process.env.BIGIQ_USERNAME || 'admin';
+const BIGIQ_PASSWORD: string = process.env.BIGIQ_PASSWORD || 'admin';
+const BIGIQ_POOL: string = process.env.BIGIQ_POOL || 'unknown';
```

Comment:
Throw exception here instead of giving a default value. 
With the given default value, ADCaaS may run into uncontrollable state.

Reply:
The default values themselves is harmless. If we hope to ensure every required settings are there, we will need to have another piece of code to do this check when the program startup. However, we do not have that code now. Need to do that in another change set.